### DISCUSSION TITLE
Duda sobre el schema instance formato fecha

### DISCUSSION THREAD

**Discussion by NicolasPLG** [Cotalker/documentation#804] [2. Questions and Answers]
Hola a todos

Les escribo porque encontré que hay un problema, no sé si nivel de implementación o un bug, pero se los explicaré

1.- Nuestra área utiliza un schemainstance de fecha para los proyectos, y ésta se encuentra en el asset tasks de nuestra sm. Ese campo de fecha se actualiza cuando levantamos un formulario que contiene una componente "fecha" y una componente "texto". Lo actualizamos mediante un nwr jsonpatch de tipo patch con operación replace.

2.-  El code del json patch es el siguiente:

```{
                    "_id" : ObjectId("6241bfea8be58e00079827c9"), 
                    "start" : "enviar_mensaje_exito", 
                    "maxIterations" : NumberInt(5), 
                    "stages" : [
                        {
                            "_id" : ObjectId("6241be5241e866000792313c"), 
                            "key" : "patch_property", 
                            "name" : "NWRequest", 
                            "version" : "2.0.0", 
                            "data" : {
                                "url" : "$JOIN##(http://core.production:9000/api/v2/properties/jsonpatch/)#($VALUE#asset)", 
                                "method" : "patch", 
                                "headers" : {
                                    "admin" : "true", 
                                    "Authorization" : "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI1Y2NiMGQ0ZGVjZjZlYjVmMzQ2MTRhOTciLCJyb2xlIjoidXNlciIsImNvbXBhbnkiOiI1OGZlODNlZGE4YjcyNTk2MjViOWIxNWQiLCJkYXRlIjoxNjI0NDE3MzI2MjIwLCJpYXQiOjE2MjQ0MTczMjZ9.MZH_FTtV-dWW3sY8QSq-eg06GyP9s_xjfWCB816l4yM"
                                }, 
                                "defaultAuth" : false, 
                                "body" : [
                                    {
                                        "op" : "replace", 
                                        "path" : "/schemaInstance/fecha_kickoff", 
                                        "value" : "$VALUE#sentAnswer|data|[find=>identifier=fecha_kickoff_ops]|process|0|[date=>format=YYYY-MM-DD@America/Santiago]"
                                    }
                                ]
                            }, 
                            "next" : {
                                "SUCCESS" : "", 
                                "ERROR" : ""
                            }, 
                            "customNetworkRequest" : [

                            ]
                        }, 
```

3.- Se ejecuta de manera correcta el patch, pero está el detalle que se actualiza con un día anterior, como se muestra en la siguiente imagen:
![image](https://user-images.githubusercontent.com/61419471/160482797-57c28cd0-056b-42c4-8c7a-98bde167069d.png)

4.- El formato que se requiere en el schemainstance es YYYY-MM-DD, ya que al dejarlo DD-MM-YYYY, no lo tomará de forma correcta.

5.- Entonces lo levanto para saber:

5.1.- Si es un error de implementación, que me indiquen qué se debe colocar

5.2.- Si no es el comportamiento esperado de la actualización en el esquema lo levanto como un bug.

Gracias :)

**Comment by acedward** [comment]
Podría ser un problema de zona horaria.

Una fecha a las 00:01 del día en una zona horaria podría ser el día anterior en otra.

**Reply by NicolasPLG** [reply]
ENtonces en teoria, cuando se atrase el reloj, debería funcionar de mejor manera?
